calculator.go

```diff
@@ -1,13 +1,111 @@
-// Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"math"
+	"strconv"
+	"strings"
+)
+
+// Add takes two or more float64 values and returns their sum.
+func Add(a float64, b float64, nums ...float64) float64 {
+	sum := a + b
+	for _, n := range nums {
+		sum += n
+	}
+	return sum
+}
+
+// Subtract takes a variable number of float64 and returns their difference.
+func Subtract(a float64, b float64, nums ...float64) float64 {
+	difference := a - b
```

Comment:
Is this the right variable name any more? Does it make sense to ask what the "difference" of 1, 2, and 3 is? I'm not sure it does, or, at least, _I_ know what you mean, but we can clarify this a little.

Reply:
This might be an American thing, but I was taught that the formal name for any number of subtractions is "difference" (https://en.wikipedia.org/wiki/Subtraction#:~:text=The%20result%20of%20a%20subtraction,by%20the%20minus%20sign%2C%20%E2%88%92.&text=Because%200%20is%20the%20additive,such%20as%20addition%20and%20multiplication.). I'm happy to use another variable name that might be more universally applicable if you have any thoughts

Reply:
This is an interesting question! 'Subtraction' in general always refers to two numbers, as in the Wikipedia article, so 'difference' completely makes sense in that case. If you were to ask the average person "What is the difference of 1, 2, and 3?" I'm not sure they'd know what you mean. Or they might be able to _guess_ what you meant (subtract the sum of 2 and 3 from 1), but feel that it was inelegantly phrased.

How about `result`?